h2o-py/h2o/grid/infogram_grid.py

```diff
@@ -0,0 +1,44 @@
+import h2o
+
+
+def _train_and_get_models(model_class, x, y, train, **kwargs):
+    from h2o.automl import H2OAutoML
+    from h2o.grid import H2OGridSearch
+
+    model = model_class(**kwargs)
+    model.train(x, y, train)
+    if model_class is H2OAutoML:
+        return [h2o.get_model(m[0]) for m in model.leaderboard["model_id"].as_data_frame(False, False)]
+    elif model_class is H2OGridSearch:
+        return [h2o.get_model(m) for m in model.model_ids]
+    else:
+        return [model]
+
+
+def infogram_grid(infogram, model_class, y, training_frame, **kwargs):
+    """
+    Runs grid over different feature subsets selected by infogram.
+
+    :param infogram: H2OInfogram object
+    :param model_class: H2O Estimator class, H2OAutoML, or H2OGridSearch
+    :param y: response column
+    :param training_frame: training frame
+    :param kwargs: Arguments passed to the constructor of the model_class
+    :return: list of H2O models
+    """
+    from h2o import H2OFrame
+    from h2o.estimators import H2OInfogram
+    from h2o.utils.typechecks import assert_is_type
+
+    assert_is_type(infogram, H2OInfogram)
+    assert hasattr(model_class, "train")
+    assert_is_type(y, str)
+    assert_is_type(training_frame, H2OFrame)
+
+    safety = infogram.get_admissible_score_frame().sort("safety_index", False)
+    cols = [x[0] for x in safety["column"].as_data_frame(False, False)]
+    subsets = [cols[0:i] for i in range(1, len(cols)+1)]
+    models = []
+    for x in subsets:
+        models.extend(_train_and_get_models(model_class, x, y, training_frame, **kwargs))
```

Comment:
so if I have let's say 10 subsets, it will train 10 `model_class` models, which means potentially 10 AutoML runs or 10 grids, each potentially generating dozen models or more, and then everything gets merged together in the same list of models...
Let's put aside the problem of having AutoML/Grids trained in a loop, shouldn't the models be at least grouped by subset to make it easier for the end user to figure out—by herself if she wants to—which subset fits her needs?

Reply:
I am not sure if that's a real use-case. When the user runs the infogram grid they want to get some best model with respect to some criteria so I think the use would be more like:
1. run infogram grid
2. calculate fairness metrics
3. pick some set of best models based on performance/fairness metrics
4. use some other criteria to pick the overall best model, here I would use the number of features etc.

But I can make it a dictionary. Do you think `{number_of_features : list_of_models}` is sufficient or should I use `{ concatenated_feature_names : list_of_models}`? Both versions would have unique mapping to the subsets but the latter is more explicit but can get very long with a lot of features...